HingeEmbeddingCriterion.lua

```diff
@@ -2,24 +2,38 @@ local HingeEmbeddingCriterion, parent = torch.class('nn.HingeEmbeddingCriterion'
 
 function HingeEmbeddingCriterion:__init(margin)
    parent.__init(self)
-   margin = margin or 1 
-   self.margin = margin 
-   self.gradInput = torch.Tensor(1)
+   self.margin = margin or 1
+   self.sizeAverage = true
 end 
  
 function HingeEmbeddingCriterion:updateOutput(input,y)
-   self.output = input[1]
-   if y == -1 then
-	   self.output = math.max(0,self.margin - self.output);
+   self.buffer = self.buffer or input.new()
+   if not torch.isTensor(y) then self.ty = self.ty or input.new():resize(1); self.ty[1]=y; y=self.ty end
+
+   self.buffer:resizeAs(input):copy(input)
+   self.buffer[torch.eq(y, -1)] = 0
```

Comment:
Thanks for the comment, your right. This assumes that the memory manager does a bad job though, is it really like that? Nevertheless, do you know how to change that without using 2 more buffers?